bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/thing/ThingTypeResource.java

```diff
@@ -125,6 +139,26 @@ public Response getByUID(@PathParam("thingTypeUID") @ApiParam(value = "thingType
         }
     }
 
+    @GET
+    @Path("/{thingTypeUID}/firmwares")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Get all available firmwares for provided thingType", response = StrippedThingTypeDTO.class, responseContainer = "Set")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "OK"),
+            @ApiResponse(code = 204, message = "no firmwares found") })
+    public Response getFirmwares(@PathParam("thingTypeUID") @ApiParam(value = "thingTypeUID") String thingTypeUID,
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @ApiParam(value = HttpHeaders.ACCEPT_LANGUAGE) String language) {
+        ThingTypeUID athingTypeUID = new ThingTypeUID(thingTypeUID);
+        if (firmwareRegistry != null) {
```

Comment:
this check is not needed.... there is a filter implemented that checks all dependencies